### DISCUSSION TITLE
Several builds fail from missing header files

### DISCUSSION THREAD

**Discussion by jacktose** [Homebrew/discussions#3462] [Linux] [answered]
I have several formulae (below) that are failing to build on upgrade, with complaints about missing header files, including `limits.h` and `errno.h`. `xz` is a good example because it has no dependencies.
```
libidn2 (2.3.2) < 2.3.3
openldap (2.6.2) < 2.6.3
openssl@1.1 (1.1.1p) < 1.1.1q
pipenv (2022.6.7) < 2022.7.24
sqlite (3.39.1) < 3.39.2
vim (9.0.0000) < 9.0.0050
xz (5.2.5) < 5.2.5_1
```

[gist with examples](https://gist.github.com/jacktose/170658d03ed73a936f89955fbade40d6/)

I'm on CentOS 7 with permissions only for my userspace, hence the builds. `gcc` seems to be properly poured (see gist) but a minimal compilation with just `#include <limits.h>` fails to find that file.

Can someone point me in the right direction to resolve this? Should I build `gcc` from source? Install `gcc@5` or some other version? Install some `linux-headers@X`? I'm a noob with compilers, so any advice is appreciated.

**Comment by jacktose** [comment]
The problem appears to be solved by the installation of `linux-headers@4.4`. IIRC, it was auto-installed to build `glibc`, but I had removed it because it was not a (~non-build~ runtime) requirement for any of my formulae. But if it's functionally necessary to build the above formulae, shouldn't that be reflected in their dependencies?

**Reply by SMillerDev** [reply]
It should be. I guess your system headers are older than the ones on CI though, so that's why it wasn't detected. Could you make a PR to adjust the glib formula to include this?

**Reply by jacktose** [reply]
Sure thing. Here's a stab at it: https://github.com/Homebrew/homebrew-core/pull/106680

**Comment by carlocab** [comment]
@danielnachun will know more about this. CC @Homebrew/linux

**Reply by danielnachun** [reply]
I think we could consider making `linux-headers@4.4` a run time dependency of `glibc` on the grounds that users who need brewed `glibc` are very likely to also need newer Linux headers.  Let's see if any other Linux maintainers feel similarly.

**Reply by jacktose** [reply]
Per [suggestion](https://github.com/orgs/Homebrew/discussions/3462#discussioncomment-3228480), here's a PR for that: https://github.com/Homebrew/homebrew-core/pull/106680

**Comment by jacktose** [comment, marked as answer]
This was fixed by work done in Homebrew/homebrew-core#106837
